src/internal/ucx/registration_resource.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "internal/ucx/registration_cache.hpp"
+
+#include "srf/memory/adaptors.hpp"
+
+#include <memory>
+
+namespace srf::internal::ucx {
+
```

Comment:
Could use a top level block describing function and purpose.

Reply:
I'm going to resolve these as I update the next comment.